### ISSUE TITLE
export onnx  with nms  failed

### ISSUE THREAD

**Issue by 4399123** [ultralytics/ultralytics#20047]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

I use the pt model of yolov11 to convert to the onnx model. When nms=False, the conversion can be normal, but when nms=True, an error will be reported [Floating point exception]

![Image](https://github.com/user-attachments/assets/ab26aae9-8b1a-4796-b20a-669c341d1f6c)

### Environment

Ultralytics 8.3.99 � Python-3.8.13 torch-1.12.1 

### Minimal Reproducible Example

from ultralytics import YOLO
import os

# Load a model
model = YOLO("pt/11_n.pt")

model.export(format="onnx",imgsz=640,simplify=True,batch=1,device="0",nms=True)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @4399123, thank you for reaching out and for using Ultralytics 🚀! We appreciate your detailed report. An Ultralytics engineer will review and assist you shortly.

In the meantime, here are some steps to help us better understand and resolve the issue:

- If this is a 🐛 Bug Report, you've already provided a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/)—thank you! Please ensure you've tested on the latest version of the `ultralytics` package as outlined below.
- If this is related to custom training, it would be helpful to share any additional logs or relevant dataset details.

## Upgrade

Ensure you're using the latest `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to confirm the issue persists in the latest version:

```bash
pip install -U ultralytics
```

## Environments

If you continue to encounter issues, consider running YOLO in one of the following verified environments with all dependencies preinstalled:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Feel free to also explore our [Docs](https://docs.ultralytics.com/) for additional resources or join the Ultralytics community for further discussions:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) for community insights

We appreciate your patience and cooperation 😊.

**Comment by Y-T-G** [comment]
Your ONNX version is really old. You can try updating:

`pip install -U onnx`

**Comment by Y-T-G** [comment]
You should also provide the output of `yolo checks` which is required for the bug report and mentioned in the report template.

**Comment by 4399123** [comment]
Ultralytics 8.3.99 🚀 Python-3.8.5 torch-1.12.1+cu116 CUDA:0 (NVIDIA A40, 45416MiB)
YOLO11n summary (fused): 100 layers, 2,582,347 parameters, 0 gradients

PyTorch: starting from 'pt/11_n.pt' with input shape (1, 3, 640, 640) BCHW and output shape(s) (1, 300, 6) (5.2 MB)

ONNX: starting export with onnx 1.17.0 opset 15...
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
Floating point exception

**Comment by 4399123** [comment]
I have updated onnx to the latest version, but it still doesn't work

**Comment by 4399123** [comment]
![Image](https://github.com/user-attachments/assets/a3a37ef7-531a-473b-b121-6512deb841e0)

**Comment by glenn-jocher** [comment]
Thanks for the update and for trying the ONNX upgrade.

Could you try updating PyTorch to a more recent version? Version 1.12.1 might have compatibility issues with ONNX NMS export. You can update using `pip install -U torch torchvision torchaudio`.

Also, please try exporting the official `yolo11n.pt` model with `nms=True` to see if the issue persists with a standard model.
